src/track/playcounter.h

```diff
@@ -1,56 +1,94 @@
-#ifndef MIXXX_PLAYCOUNTER_H
-#define MIXXX_PLAYCOUNTER_H
+#pragma once
 
-#include "util/assert.h"
+#include <QDateTime>
+#include <QMetaType>
+#include <QtDebug>
 
+#include "util/assert.h"
 
-// Counts the total number of times a track has been played
-// and if the track has been played during the current session.
-class PlayCounter {
-public:
-    explicit PlayCounter(int timesPlayed = 0):
-        m_iTimesPlayed(timesPlayed),
-        m_bPlayed(false) {
+/// Counts the total number of times a track has been played
+/// and if the track has been played during the current session.
+class PlayCounter final {
```

Comment:
why final?

Reply:
Otherwise the destructor would need to be virtual. But there is no reason for such extensibility.